store/store.go

```diff
@@ -20,11 +20,17 @@ var (
 type Store interface {
 	Save(d interface{}) error
 	Update(selector interface{}, d interface{}) error
-	Delete(find Fields) error
-	Read(find Fields, remove Fields, result interface{}) error
-	ReadAll(find Fields, remove Fields) Iterator
+	Delete(id Field) error
```

Comment:
For any of these `Fields` we should have the interface be as specific as possible.  For example, I suspect id will always be a string, so that's what the argument type should be.